index.js

```diff
@@ -257,21 +257,39 @@ FastForEach.prototype.added = function (changeItem) {
     var childNodes = ko.virtualElements.childNodes(templateClone);
     // Note discussion at https://github.com/angular/angular.js/issues/7851
     allChildNodes.push.apply(allChildNodes, Array.prototype.slice.call(childNodes));
-    this.lastNodesList.splice(index + i, 0, childNodes[childNodes.length - 1]);
+    this.firstLastNodesList.splice(index + i, 0, { first: childNodes[0], last: childNodes[childNodes.length - 1] });
   }
 
   if (typeof this.afterAdd === 'function') {
     this.afterAdd({
       nodeOrArrayInserted: this.insertAllAfter(allChildNodes, referenceElement),
-      foreachInstance: this}
+      foreachInstance: this
+    }
     );
   } else {
     this.insertAllAfter(allChildNodes, referenceElement);
   }
 };
 
+FastForEach.prototype.getNodesForIndex = function (index) {
+  var result = [],
+    ptr = this.firstLastNodesList[index].first,
+    last = this.firstLastNodesList[index].last;
+  result.push(ptr);
+  while (ptr && ptr !== last) {
+    ptr = ptr.nextSibling;
+    result.push(ptr);
+  }
```

Comment:
I guess it would be needed because when ptr is the last, it won't add it due to the loop condition. So after all it would be quite similar, with only a "shift" difference I think.